server/src/main/java/org/elasticsearch/action/support/IndicesOptions.java

```diff
@@ -22,138 +24,425 @@
+             */
+            public Builder resolveAliases(boolean resolveAliases) {
+                this.resolveAliases = resolveAliases;
+                return this;
+            }
+
+            /**
+             * If true, when any of the expressions does not match any indices, we consider the result of this expression
+             * empty; if all the expressions are empty then we have a successful but empty response.
+             * If false, we throw an error immediately, so even if other expressions would result into indices the response
+             * will contain only the error. Defaults to true.
+             */
+            public Builder allowEmptyExpressions(boolean allowEmptyExpressions) {
+                this.allowEmptyExpressions = allowEmptyExpressions;
+                return this;
+            }
+
+            /**
+             * Disables expanding wildcards.
+             */
+            public Builder none() {
+                matchOpen = false;
+                matchClosed = false;
+                includeHidden = false;
+                return this;
+            }
+
+            /**
+             * Maximises the resolution of indices, we will match open, closed and hidden targets.
+             */
+            public Builder all() {
+                matchOpen = true;
+                matchClosed = true;
+                includeHidden = true;
+                return this;
+            }
+
+            /**
+             * Parses the list of wildcard states to expand as provided by the user.
+             * Logs a warning when the option 'none' is used along with other options because the position in the list
+             * changes the outcome.
+             */
+            public Builder expandStates(String[] expandStates) {
+                // Calling none() simulates a user providing an empty set of states
+                none();
+                for (String expandState : expandStates) {
+                    switch (expandState) {
+                        case "open" -> matchOpen(true);
+                        case "closed" -> matchClosed(true);
+                        case "hidden" -> includeHidden(true);
+                        case "all" -> all();
+                        case "none" -> {
+                            none();
+                            if (expandStates.length > 1) {
+                                DEPRECATION_LOGGER.warn(DeprecationCategory.API, "expand_wildcards", WILDCARD_NONE_DEPRECATION_MESSAGE);
+                            }
+                        }
+                    }
+                }
+                return this;
             }
+
+            public WildcardOptions build() {
+                return new WildcardOptions(matchOpen, matchClosed, includeHidden, resolveAliases, allowEmptyExpressions);
+            }
+        }
+
+        public static Builder newBuilder() {
+            return new Builder();
+        }
```

Comment:
Same comment as before about going from `newBuilder()` -> `builder()` for consistency